### PR TITLE
Keyboard-driven inputs (2)

### PR DESCRIPTION
**Pull request by BurningTreeC**
This PR is the same as #4458 (which has too many merge conflicts)

### CODE REVIEW

**Comment by BurningTreeC** [comment]
[The Demo](http://tag-picker-kb.tiddlyspot.com/) is up to date

**Comment by Jermolene** [comment]
Hi @BurningTreeC apologies for the delay, I've just tried the demo and it's looking great. I wondered if we might implement keyboard handling for the search dropdown in the menubar as well?

**Comment by pmario** [comment]
>I wondered if we might implement keyboard handling for the search dropdown in the menubar as well?

I think ... yes. ... But shouldn't it be a new PR?

**Comment by BurningTreeC** [comment]
@Jermolene @pmario - yes we can implement keyboard handling for the menubar search dropdown as well and I agree that it should be a new PR. I'll prepare a PR when this one is merged, is that OK for you?

**Comment by saqimtiaz** [comment]
@BurningTreeC @Jermolene I have reviewed all changes since we last discussed this. Overall I am very happy with this and think we should merge it sooner rather than later, so that we can get more testing done via the pre-release.

@BurningTreeC the only further suggestion I have is to perhaps rename filter, filter2 and list1, list 2 to primaryListFilter, secondaryListFilter, primaryList and secondaryList respectively. These names would probably aid understanding for others trying to read and adapt the macro to their own needs.

**Comment by BurningTreeC** [comment]
Thanks @saqimtiaz, I'll make those changes ASAP

**Comment by BurningTreeC** [comment]
>the only further suggestion I have is to perhaps rename filter1, filter2 and list1, list2 to primaryListFilter, secondaryListFilter, primaryList and secondaryList respectively. These names would probably aid understanding for others trying to read and adapt the macro to their own needs.

I've made those updates and the Demo is up to date, too

**Comment by saqimtiaz** [comment]
@BurningTreeC @Jermolene I am happy with how this has shaped up and think it can be merged.

**Comment by twMat** [comment]
I have now waited for [over two minutes](https://groups.google.com/d/msg/tiddlywikidev/_ovhxaQbWms/IZ5zev2BBwAJ) for this to get merged. I can't take it much longer.  ;-) ;-) ;-)

File: core/modules/editor/engines/framed.js [hunk 1]
```diff
@@ -108,13 +108,20 @@ Set the text of the engine if it doesn't currently have focus
...
 		}
 		// Fix the height if needed
 		this.fixHeight();
 	}
 };
 
+/*
+Update the DomNode with the new text
+*/
+FramedEngine.prototype.udpateDomNodeText = function(text) {
```
Comment (saqimtiaz) [inline core/modules/editor/engines/framed.js:121]: typo, updateDomNodeText

Comment (BurningTreeC) [reply core/modules/editor/engines/framed.js:121]: right, thanks @saqimtiaz

Comment (BurningTreeC) [reply core/modules/editor/engines/framed.js]: resolved

**Comment by BurningTreeC** [comment]
The Demo is up to date with the latest typo-fix

**Comment by BurningTreeC** [comment]
@Jermolene, this would be ready to merge. Is there anything holding this back?

**Comment by Jermolene** [comment]
Thanks @BurningTreeC and apologies for the delay.

**Comment by saqimtiaz** [comment]
@BurningTreeC @Jermolene 

I've just discovered a significant issue with the way we have implemented this.

 It requires both the keyboard driven input macro that creates the search input, and the code handling displaying the search results to use the same filter to return search results.

However, the search results can be customised:
https://tiddlywiki.com/#Customising%20search%20results

With an alternative $:/tags/SearchResults tiddler like the one provided in the example at the link above, the keyboard navigation is completely broken since the keyboard driven input and the search results display code no longer use the same filter.

I haven't had the opportunity to give this much thought, but the answer might lie in storing the search filter to be used in a common location that can be accessed by both the macro and the search results code.

**Pull request merged**